server/apps/dictionaries/service.py

```diff
@@ -10,34 +10,57 @@
 
 import re
 import logging
-from flask import request, json
+import collections
+from flask import json
 from superdesk.services import BaseService
 from superdesk.errors import SuperdeskApiError
-from superdesk import get_resource_service
 from apps.dictionaries.resource import DICTIONARY_FILE
-from eve import ETAG
 
 
 logger = logging.getLogger(__name__)
 
 
+def encode_dict(words_dict):
+    return json.dumps(words_dict)
+
+
+def decode_dict(words_list):
+    return json.loads(words_list)
+
+
+def train(features):
+    model = collections.defaultdict(lambda: 1)
+    for f in features:
+        model[f] += 1
+    return model
+
+
 def words(text):
     return re.findall('[a-z]+', text.lower())
```

Comment:
Better to use `\w` instead of `[a-z]` IMO, because we deal with unicode text and we want to recognize the words containing non-ascii letters, too.